packages/block-library/src/embed/util.js

```diff
@@ -95,11 +101,16 @@ export const createUpgradedEmbedBlock = (
 
 	const matchedBlock = findMoreSuitableBlock( url );
 
+	const WP_VARIATION = getBlockVariations( DEFAULT_EMBED_BLOCK )?.find(
+		( { name } ) => name === 'wordpress'
+	);
+
 	// WordPress blocks can work on multiple sites, and so don't have patterns,
 	// so if we're in a WordPress block, assume the user has chosen it for a WordPress URL.
 	const isCurrentBlockWP =
-		providerNameSlug === WP_VARIATION.attributes.providerNameSlug ||
-		type === WP_EMBED_TYPE;
+		WP_VARIATION && // Is the WordPress variation of the embed block registered?
```

Comment:
This comment/question feels like a personal note. Can you rephrase it to something like: `Check if .... is registered` ?

Reply:
Can do! I tend to leave out the `Check if ... is` part when commenting conditionals (since it takes up space), but I don't mind adding it back.